File: 20922/20922.cpp
#include <iostream>
using namespace std;
inline int max(int a,int b){return a > b? a:b;}

int main()
{
    int N, K, A[200001] = {0,}, cnt[100001] = {0,};
    scanf("%d %d", &N, &K);
    for (int i = 0; i < N; i++) scanf("%d", &A[i]);
    
    int s = 0, e = 0, ret = 0;
    while (e < N)
    {
        while (s < e && cnt[A[e]] == K)
        {
            cnt[A[s]]--;
            s++;
        }
        ret = max(ret,e-s+1);
        cnt[A[e]]++;
        e++;
    }
    printf("%d \n", ret);
    return 0;
}

File: 20922/20922.py
import sys
input = sys.stdin.readline
N,K = map(int,input().rstrip().split())
A = list(map(int,input().rstrip().split()))
count = [0 for _ in range(100001)]
s,e,ret = 0,0,0
while e < N:
    while s < e and count[A[e]] == K:
        count[A[s]] -= 1
        s += 1
    count[A[e]] += 1
    ret = max(ret, e-s+1)
    e += 1
print(ret)